cmd/kube-apiserver/app/server.go

```diff
@@ -245,12 +245,13 @@ func (s *APIServer) verifyClusterIPFlags() {
 	}
 }
 
-func newEtcd(etcdConfigFile string, etcdServerList []string, interfacesFunc meta.VersionInterfacesFunc, defaultVersion, storageVersion, pathPrefix string) (etcdStorage storage.Interface, err error) {
+func newEtcd(etcdConfigFile string, etcdServerList []string, interfacesFunc meta.VersionInterfacesFunc, defaultVersion, storageVersion, pathPrefix string) (storage.Interface, string, error) {
```

Comment:
Let's:
* Not remove the return parameter names; they help readability significantly.
* Not default storage version to default version; instead, return an error if storage version is ""
* Default version isn't even used and is confusing; remove it as a parameter